### PR TITLE
Improvement: Binance Futures rate limits

### PR DESCRIPTION
**Pull request by gloriousCode**
# PR Description
Adds rate limits for Binance futures endpoints.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Travis with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/binance/binance_cfutures.go [hunk 1]
```diff
@@ -741,14 +805,16 @@ func (b *Binance) GetFuturesOrderbookTicker(symbol currency.Pair, pair string) (
 	if pair != "" {
 		params.Set("pair", pair)
 	}
-	return resp, b.SendHTTPRequest(exchange.RestCoinMargined, cfuturesSymbolOrderbook+params.Encode(), limitDefault, &resp)
+	return resp, b.SendHTTPRequest(exchange.RestCoinMargined, cfuturesSymbolOrderbook+params.Encode(), rateLimit, &resp)
 }
 
 // GetFuturesLiquidationOrders gets orderbook ticker for symbol
```
Comment (shazbert) [inline exchanges/binance/binance_cfutures.go:811]: returns forced liquidations not orderbook

**Review by shazbert — CHANGES_REQUESTED** [review]
All spot pairs enabled on REST fetching:
[DEBUG]|ORDER| 26/02/2021 11:33:35 |Order manager: Processing orders for exchange Binance. - deadline would be exceeded by retry, status: 429 Too Many Requests

Out of Scope Issues: 
|TICKER| 26/02/2021 11:33:25 |Failed to get REST ticker. Error: no tickers associated with base currency DASH - There might be an issue with loading tickers in the ticker package?
[ERROR]|LOG| 26/02/2021 11:32:42 |Binance failed to get cryptocurrency deposit addresses. Err: -1021=Timestamp for this request is outside of the recvWindow.

File: exchanges/binance/ratelimit.go [hunk 2]
```diff
@@ -11,63 +11,188 @@ const (
...
+		uFuturesGetAllOrdersRate:
+		limiter, tokens = r.UFuturesOrdersRate, 5
+	case uFuturesCountdownCancelRate:
+		limiter, tokens = r.UFuturesOrdersRate, 10
+	case uFuturesCurrencyForceOrdersRate,
+		uFuturesSymbolOrdersRate:
+		limiter, tokens = r.UFuturesOrdersRate, 20
+	case uFuturesIncomeHistoryRate:
+		limiter, tokens = r.UFuturesOrdersRate, 30
+	case uFuturesPairOrdersRate,
```
Comment (shazbert) [inline exchanges/binance/ratelimit.go:159]: For some reason from our order manager calling (*Binance).UAllAccountOpenOrders we are getting a 429 error. I cannot seem to find the reason why the weighting seems correct. This is the only issue I have encountered between USDT and COIN futures with all currency pairs enabled. Spot disabled.

**Review by shazbert — CHANGES_REQUESTED** [review]
Actual in scope nit. Everything else seems to be working correctly.

File: exchanges/binance/ratelimit.go [hunk 3]
```diff
@@ -11,63 +11,188 @@ const (
...
+	uFuturesBatchOrdersRate
+	uFuturesGetAllOpenOrdersRate
+	cFuturesDefaultRate
+	cFuturesHistoricalTradesRate
+	cFuturesTickerPriceHistoryRate
+	cFuturesIncomeHistoryRate
+	cFuturesOrderbook50Rate
+	cFuturesOrderbook100Rate
+	cFuturesOrderbook500Rate
+	cFuturesOrderbook1000Rate
```
Comment (MadCozBadd) [inline exchanges/binance/ratelimit.go:74]: These have not been used

File: exchanges/binance/ratelimit.go [hunk 4]
```diff
@@ -11,63 +11,188 @@ const (
...
+	uFuturesKlineMaxRate
+	uFuturesTickerPriceHistoryRate
+	uFuturesOrdersDefaultRate
+	uFuturesGetAllOrdersRate
+	uFuturesAccountInformationRate
+	uFuturesOrderbookTickerAllRate
+	uFuturesCountdownCancelRate
+	uFuturesBatchOrdersRate
+	uFuturesGetAllOpenOrdersRate
+	cFuturesDefaultRate
```
Comment (MadCozBadd) [inline exchanges/binance/ratelimit.go:67]: This hasn't been used either so it defaults to the spot rate instead

**Review by MadCozBadd — COMMENTED** [review]
Just 2 small oopsies, otherwise a great pr!

**Comment by gloriousCode** [comment]
> Binance failed to get cryptocurrency deposit addresses. Err: -1021=Timestamp for this request is outside of the recvWindow.

I've added a larger `recvWindow` on retrieving deposit address.
For you other notes, I can enable all currencies and not experience 429s now

**Review by MadCozBadd — APPROVED** [review]
tested and works well!!!!!

**Review by shazbert — APPROVED** [review]
tACK! Thanks for addressing nits!

**Review by thrasher- — APPROVED** [review]
tACK, 2x ticker/orderbook sync runs with 2660 pairs and asset types enabled. Was not given a 418 / 429 error. Thanks for going through the tedious effort to add in rate limit for all API endpoints (spot/margin/futures)!

**Pull request merged**